### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by dirkschumacher**
This PR tries to improve the Readme a bit by giving more context around nextroute itself, features, and license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,68 @@
...
+applications across [all platforms that Go (cross)compiles
+to](https://go.dev/doc/install/source#environment).
+
+Our goal isn't to compete on specific VRP type benchmarks, but to provide a
+robust and versatile tool that can adapt to a variety of routing challenges.
+Whether you're optimizing the routes for a small fleet of delivery vans in a
+city or managing complex logistics for a global supply chain, nextroute is
+equipped to help you find efficient solutions.
+
+## Features
```
Comment (merschformann) [inline README.md:15]: While I totally like it, this introduces another place that we need to maintain / update when we make changes. I am merely pointing it out though, I don't have a strong opinion on this. :blush:

**Review by merschformann — APPROVED** [review]
I like it, thanks! I am not sure about the list maintenance overhead vs. just linking docs. ¯\_(ツ)_/¯

File: README.md (hunk 1 above)
Comment (dirkschumacher) [reply README.md:15]: One of the cool things about nextroute are its many features. I think we should provide a short list in the readme to help users get feel what the package can do without clicking on another link.

Maintaining such a high-level table won't be too much extra work IMHO.

Comment (merschformann) [reply README.md:15]: Yeah fair, it should be fine. Links should be stable and we won't remove features anyway. Worst case would be that one is missing. :relaxed:

**Review by larsbeck — APPROVED** [review]

**Pull request merged**